lib/jormungandr/src/Cardano/Wallet/Jormungandr/Network.hs

```diff
@@ -64,7 +66,18 @@ import Servant.Client.Core
 import Servant.Links
     ( Link, safeLink )
 
--- TODO: Implement a NetworkLayer
+mkNetworkLayer :: MonadIO m => JormungandrLayer m -> NetworkLayer t m
+mkNetworkLayer j = NetworkLayer
+    { networkTip = do
+        t@(BlockId hash) <- lift $ getTipId j
+        lift $ getBlock j t >>= \case
+            (Just b) -> return (hash, header b)
+            Nothing -> liftIO . throwIO $ ErrGetBlockNotFound t
```

Comment:
Could wrap this in some `ErrInternalPanicOrSomething`-wrapper… or handle the error entirely differently

Reply:
I think adding another constructor to `ErrNetworkTip` would be fine.
`networkTip` is used to find out the tip slot id.